presto-main/src/main/java/com/facebook/presto/operator/scalar/IpPrefixFunctions.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.LiteralParameters;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.google.common.net.InetAddresses;
+import io.airlift.slice.Slice;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.type.IpAddressOperators.castFromVarcharToIpAddress;
+import static com.facebook.presto.type.IpPrefixOperators.castFromVarcharToIpPrefix;
+import static com.facebook.presto.util.Failures.checkCondition;
+import static io.airlift.slice.Slices.utf8Slice;
+
+public final class IpPrefixFunctions
+{
+    private IpPrefixFunctions() {}
+
+    @Description("IP prefix for a given IP address and subnet size")
+    @ScalarFunction("ip_prefix")
+    @SqlType(StandardTypes.IPPREFIX)
+    public static Slice ipPrefix(@SqlType(StandardTypes.IPADDRESS) Slice value, @SqlType(StandardTypes.BIGINT) long subnetSize)
+    {
+        byte[] ipV4Map = new byte[12];
```

Comment:
You can check whether it's a v4 or v6 address after converting to InetAddress. It's easier to read that way.